tools/ci/get-ci-matrix.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+#
+
+# This script counts the total number of crates in the SDK (including examples),
+# and creates a set of batches based on that size in a format that GitHub Actions
+# can consume as a matrix strategy.
+
+import json
+import math
+import os
+import shlex
+import subprocess
+import sys
+import unittest
+
+DESIRED_WORKERS = 6
+
+
+def main():
+    if len(sys.argv) == 1:
+        print(f"Usage: {sys.argv[0]} <rust versions to use...>")
+        return 1
+
+    rust_versions = sys.argv[1:]
+    repository_root = get_cmd_output("git rev-parse --show-toplevel")
+    os.chdir(repository_root)
+
+    crate_paths = get_cmd_output("./tools/ci/list-sdk-crate-paths.sh").splitlines(keepends=False)
```

Comment:
would probably keep this all in-script unless you have a compelling reason for it to be separate (`os.listdir`, etc.)

Reply:
Yeah, the separate scripts are an artifact of my incremental approach to solving the problem. I should clean it up!